astropy/time/core.py

```diff
@@ -777,7 +777,8 @@ def __setitem__(self, item, value):
             if hasattr(self, attr):
                 delattr(self, attr)
 
-        if value in (np.ma.masked, np.nan):
+        if (value is np.ma.masked or value is np.nan or
+                (getattr(value, 'ndim', 0) == 0 and value == np.nan)):
```

Comment:
I think this should be `.. and np.isnan(value)` as `np.array(np.nan) == np.nan` gives `False`.

Reply:
That's what one gets for pushing something up without having time to test... Sigh... But `np.isnan` also doesn't work since that chokes on strings. And then I wondered about arrays, which made me think that perhaps the test for `nan` was not actually needed. And at least all tests pass without it, so that's what I did now.